ethcore/light/src/net/mod.rs

```diff
@@ -335,6 +345,40 @@ mod id_guard {
 	}
 }
 
+/// Provides various statistics that could
+/// be used to compute costs
+struct Statistics {
+	/// Samples of peer count
+	peer_counts: VecDeque<usize>,
+}
+
+impl Statistics {
+	pub fn new() -> Self {
+		Statistics {
+			peer_counts: VecDeque::with_capacity(MOVING_SAMPLE_SIZE),
+		}
+	}
+
+	/// Add a new peer_count sample
+	pub fn add_peer_count(&mut self, peer_count: usize) {
+		while self.peer_counts.len() >= MOVING_SAMPLE_SIZE {
```

Comment:
Could be simplified by using a fixed size array/vector (initialized wit 1s) and `index: usize` that wraps after the end

Reply:
Good point, but the current implementation is more accurate in case when the number of samples is less than `MOVING_SAMPLE_SIZE` (it would be full only after an hour (256 * 15 / 60)).

Reply:
Fair enough, although that could be simple solved with a flag (i.e. `is_fully_filled ? 0..len : 0..index`).